rest/route/github.go

```diff
@@ -88,19 +88,7 @@ func (gh *githubHookApi) Parse(ctx context.Context, r *http.Request) error {
 		}
 	}
 
-	gh.event, err = github.ParseWebHook(gh.eventType, body)
-	if err != nil {
-		grip.Error(message.WrapError(err, message.Fields{
-			"source":  "github hook",
-			"msg_id":  gh.msgID,
-			"event":   gh.eventType,
-			"message": "rejecting github webhook",
-		}))
-		return gimlet.ErrorResponse{
-			StatusCode: http.StatusBadRequest,
-			Message:    err.Error(),
-		}
-	}
+	gh.event, _ = github.ParseWebHook(gh.eventType, body)
```

Comment:
If we don't return an error here then we'll continue to Run() right? Which would be a problem. I would think we should still error here but not log it.

Reply:
I mean, I don't think it's *technically* a problem: Run() will skip to the end of the switch and return an empty response.
Still, I agree stylistically: it's probably best to avoid relying on this for the sake of clarity and in case Run() changes.

Perhaps the following can be a path forward: we could store the event type and body in the `githubHookApi`. In the run function we can check the eventType against a list of the events of interest (`utility.StringSliceContains(eventsOfInterestSlice, gh.eventType)`) and return 200 if we're not interested. Then we call ParseWebHook and take it from there.

Reply:
I do think this is a real error though, and returning 200 for a response we can't parse is incorrect. I would prefer to continue treating this as a real error but just not logging it back to Splunk. I don't think this issue is worth the engineering bandwidth to do anything too complicated.

Reply:
parse returns error again